### PR TITLE
feat(objects): add support for Group wikis

### PR DESCRIPTION
**Pull request by spamsch**
This pull request adds support for wikis on Group level

Closes https://github.com/python-gitlab/python-gitlab/issues/1224

### CODE REVIEW

File: docs/gl_objects/wikis.rst [hunk 1]
```diff
@@ -11,6 +11,9 @@ References
   + :class:`gitlab.v4.objects.ProjectWiki`
   + :class:`gitlab.v4.objects.ProjectWikiManager`
   + :attr:`gitlab.v4.objects.Project.wikis`
+  + :class:`gitlab.v4.objects.GroupWiki`
+  + :class:`gitlab.v4.objects.GroupWikiManager`
+  + :attr:`gitlab.v4.objects.Group.wikis`
 
 * GitLab API: https://docs.gitlab.com/ce/api/wikis.html
```
Comment (JohnVillalovos) [inline docs/gl_objects/wikis.rst:19]: Please add link to the API page:

`https://docs.gitlab.com/ee/api/group_wikis.html`

File: docs/gl_objects/wikis.rst [hunk 2]
```diff
@@ -21,6 +24,10 @@ Get the list of wiki pages for a project::
 
     pages = project.wikis.list()
 
+Get the list of wiki pages for a group::
+
+    pages = group.wikis.list()
+
 Get a single wiki page::
 
     page = project.wikis.get(page_slug)
```
Comment (JohnVillalovos) [inline docs/gl_objects/wikis.rst:33]: Can we also get an example of doing a "get" of a group wiki page?

And an example of creating a group wiki page too.

**Review by JohnVillalovos — CHANGES_REQUESTED** [review]

**Comment by JohnVillalovos** [comment]
Thanks for this PR!

I left a few comments, but overall it is looking great 👍

**Comment by max-wittig** [comment]
Would it be possible for you to add some unit tests as well for this?

**Comment by spamsch** [comment]
> Would it be possible for you to add some unit tests as well for this?

There are no unit tests for project wikis. Not sure it makes sense to have unit tests for groups only. Could think about adding functional tests. Makes sense?

**Comment by nejch** [comment]
> 
> 
> > Would it be possible for you to add some unit tests as well for this?
> 
> There are no unit tests for project wikis. Not sure it makes sense to have unit tests for groups only. Could think about adding functional tests. Makes sense?

I just triggered the workflow. Group Wikis are a Premium/EE feature so let's see if this works in the CE container 😁 And you're right, we'd need to create a new `test_wikis.py` module with a group wiki mock response for this. There are examples in other modules if the functional tests fail (if it fails, maybe add `@pytest.mark.skip` to the functional test that you added, for now)

**Comment by JohnVillalovos** [comment]
FYI: Running `tox -e black` locally should resolve the black formatting issues.

**Comment by JohnVillalovos** [comment]
Thanks a lot @spamsch 😊

Looks good to me. Personally I would like it if it was rebased and squashed. As right now there are six commits and one of them is a merge commit. I think it will make the project history be better if it was rebased and squashed.

Not sure what @nejch and @max-wittig  think.

**Comment by nejch** [comment]
> Thanks a lot @spamsch 😊
> 
> Looks good to me. Personally I would like it if it was rebased and squashed. As right now there are six commits and one of them is a merge commit. I think it will make the project history be better if it was rebased and squashed.
> 
> Not sure what @nejch and @max-wittig  think.

Ok, i can also squash on merge and add the unit tests along with project wiki tests in a follow-up. Just not from my phone on my commute right now.. :D

**Pull request merged**